### ISSUE TITLE
Support for `flake8-annotations-coverage`

### ISSUE THREAD

**Issue by szymonmaszke** [charliermarsh/ruff#3210]
Hey, we are currently using [`flake8-annotations-coverage`](https://github.com/best-doctor/flake8-annotations-coverage) in our FOSS [`py-template`](https://github.com/inovintell/py-template) project, but we would like to move to `ruff` exclusively from `flake8` + `flakeheaven`. 

This plugin basically allows you to set minimum percentage coverage for annotations and raise if it is not fulfilled.

### Error Codes

- TAE001 - Too few type annotations in file

Is support for [`flake8-annotations-coverage`](https://github.com/best-doctor/flake8-annotations-coverage) planned/would be considered? Unfortunately we cannot pick up this issue ourselves at the current time.

**Comment by szymonmaszke** [comment]
Thank you for quick reply!

> However after some thought, I realized I would just set the % to 100%. What's the argument for setting a lower minimum %, rather than just enforcing 100% and adding noqa (or annotations) where needed?

We are indeed using it like that, although for migrating existing code bases such a threshold to reach might be beneficial (leaving a side reasoning for such a decision).

> Since Ruff already has all the flake8-annotations rules prefixed with ANN, would it be better for your FOSS project to enable them all and just add more type annotations or noqa to fix any warnings?

Yes, indeed, seems more reasonable to place it under `ANN`

Regarding the last question, I think I have already answered, I see no reason for new repos other than trying to adhere to the newer standard. Another reason might be to leave aside `typinit aside as I don't find it too beneficial tbh, but it's speculation.

**Comment by JonathanPlasse** [comment]
If I had to type an old codebase, I would use `--add-noqa` with `--select=ANN` to add `noqa` for `ANN` errors.
Then you can search for files with the most `noqa`to have an indicator of the files that need it the most.

**Comment by szymonmaszke** [comment]
> If I had to type an old codebase, I would use `--add-noqa` with `--select=ANN` to add `noqa` for `ANN` errors.
> 
> Then you can search for files with the most `noqa`to have an indicator of the files that need it the most.

Also an option, we would honestly be fine with anything showing us missing annotations.

**Comment by stefanitsky** [comment]
I'm not sure if this rule can be optimally implemented in ruff, due to the fact, that the rule checks the entire file for annotations, but i don't see in the code, where this could be added.

@MichaReiser could you help us understand, where it is possible to implement it?

If done in a simple way, i think, we just can check for annotations for each function separately and implement such a rule within `ANN` rules, as @matthewlloyd suggested. I can implement that, doesn't look difficult.

What do you all think?

**Comment by szymonmaszke** [comment]
@stefanitsky agreed, it would be best as a per function/method check and telling more.

Something to keep in mind - possibility to exclude `self` and `cls`.
